frontend/src/components/stages/base_stage_editor.ts

```diff
@@ -129,16 +129,90 @@ export class BaseStageEditorComponent extends MobxLitElement {
     `;
   }
 
+  private shouldDisableWaitForAllParticipants(): boolean {
```

Comment:
Can you make this a utils function that takes in list of stages, reveal stage ID?